File: Python/RSA_PlainText_Attack.py
def power(x, y, mod):
    # Initialize result
    res = 1
    while (y > 0):
        # If y is odd, multiply x with result
        if ((y & 1) != 0):
            res = (res * x)%mod
        # y must be even now
        y = y >> 1  # y = y/2
        x = (x * x)%mod  # Change x to x^2
    return res % mod


#main

e = 29
n = 17*19
msg = 50
print("Msg ",msg)
cipher = power(msg,e,n)
print("Cipher ",cipher)

d = 2
while(True):
    new_msg = power(cipher,d,n)
    if(new_msg == msg ):
        print("new_msg ",new_msg)
        print("Private key: ",d)
        break
    d += 1
    

File: CPP/RSA_PlainText_Attack.cpp
#include <bits/stdc++.h>
using namespace std;
#define int            long long int
#define nl             "\n"


int power(int b, int p, int mod)
{
    b = b%mod;
    if(p == 0)
        return 1;
    int j = power(b,p/2,mod);
    j=(j*j)%mod;
    if(p%2==1)
        j=(j*b)%mod;
    return j;
}

int32_t main()
{
    int e = 29;
    int n = 17*19;
    int msg = 50;
    cout<<"Message: "<<msg<<endl;
    int cypher = power(msg,e,n);
    cout<<"Cypher: "<<cypher<<endl;

    int d=2;
    while(1)
    {
        int new_msg=power(cypher,d, n);
        if(new_msg==msg)
        {
            cout<<"New Message: "<<new_msg<<endl;
            cout<<"Private key: "<<d<<endl;
            break;
        }
        d++;
    }

    return 0;
}
